Detectors/TPC/qc/src/GPUErrorQA.cxx

```diff
@@ -0,0 +1,67 @@
+// Copyright 2019-2025 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+#define _USE_MATH_DEFINES
+
+// root includes
+#include "TFile.h"
+#include "TH1I.h"
+
+// o2 includes
+#include "TPCQC/GPUErrorQA.h"
+#include "GPUErrors.h"
+
+ClassImp(o2::tpc::qc::GPUErrorQA);
+
+using namespace o2::tpc::qc;
+
+//______________________________________________________________________________
+void GPUErrorQA::initializeHistograms()
+{
+  TH1::AddDirectory(false);
+
+  // 1D histogram counting all reported errors
+  mMapHist["ErrorCounter"] = std::make_unique<TH1I>("ErrorCounter", "ErrorCounter", o2::gpu::errorNames.size(), -0.5, o2::gpu::errorNames.size() - 0.5);
+  mMapHist["ErrorCounter"]->GetXaxis()->SetTitle("Error Codes");
+  mMapHist["ErrorCounter"]->GetYaxis()->SetTitle("Entries");
+  // for convienence, label each bin with the error name
+  for (size_t bin = 1; bin < mMapHist["ErrorCounter"]->GetNbinsX(); bin++) {
+    auto const& it = o2::gpu::errorNames.find(bin);
+    mMapHist["ErrorCounter"]->GetXaxis()->SetBinLabel(bin, it->second);
```

Comment:
Just in case we will have a gap in the error numbers in the future, I would propose to check if `it` is a valid error Name.